pkg/operator/controller/ingress/deployment.go

```diff
@@ -623,6 +649,12 @@ func desiredRouterDeployment(ci *operatorv1.IngressController, ingressController
 		env = append(env, corev1.EnvVar{Name: RouterDisableHTTP2EnvName, Value: "true"})
 	}
 
+	if enabled, value := HardStopAfterIsEnabled(ci, ingressConfig); enabled {
+		if _, err := time.ParseDuration(value); err != nil {
+			env = append(env, corev1.EnvVar{Name: RouterHardStopAfterEnvName, Value: value})
```

Comment:
`time.ParseDuration` can't parse things that specify days (ie `1d`). We don't expect users to be setting values that high right?

Reply:
Also if we fail to parse the annotation should we log that?

Reply:
I think the logic is inverted here:  We should use `value` only if `time.ParseDuration(value)` returns a nil error value.

Reply:
Perhaps `HardStopAfterIsEnabledByAnnotation` should perform the validation.  What should happen if `ingresses.config/cluster` has a valid annotation and then the user adds an invalid annotation to the ingresscontroller?  I think the first annotation should remain in effect, but with the current logic, the invalid annotation on the ingresscontroller causes the valid annotation on the ingress.config to be ignored.

Reply:
Done and added a test; though I will add an additional test that explicitly sets "" as a timeout value - just need to rearrange the test a bit now.